bzzeth/bzzeth.go

```diff
@@ -172,19 +173,24 @@ func (b *BzzEth) handleNewBlockHeaders(ctx context.Context, p *Peer, msg *NewBlo
 		case hdr, ok := <-deliveries:
 			if !ok {
 				p.logger.Debug("bzzeth.handleNewBlockHeaders", "delivered", deliveredCnt)
-				return
+				// todo: introduce better errors
+				return nil
 			}
 			ch := newChunk(hdr)
 			deliveredCnt++
 			p.logger.Trace("bzzeth.handleNewBlockHeaders", "hash", ch.Address().Hex(), "delivered", deliveredCnt)
+
+			req.lock.RLock()
```

Comment:
It was async before as well. I am not sure is it really a bug, but it might be potentially dangerous. Basically, if I am right, there is a lock in req that is to be used to update `hashes` map. This lock is used for update but not for read, which might not be fatal, but still a bit not correct :)